### PR TITLE
LFA-1736 Caused accounts due row to not appear if no data for it is found

### PR DESCRIPTION
**Pull request by SVarallo**
Jira ticket at https://companieshouse.atlassian.net/browse/LFA-1736

### CODE REVIEW

File: src/controllers/check.your.answers.controller.ts [hunk 1]
```diff
@@ -26,12 +27,16 @@ const route = async (req: Request, res: Response, next: NextFunction): Promise<v
       const request: IExtensionRequest = sessionService.getRequest(req.chSession);
       const fullRequest: ExtensionFullRequest =
         await apiClient.getFullRequest(companyNumber, token, request.extension_request_id);
+      let companySummaryListRows = buildCompanySummaryListRows(companyInSession, true);
+      companySummaryListRows = companySummaryListRows.concat(
```
Comment (mwestacott) [inline src/controllers/check.your.answers.controller.ts:31]: Just wonder if this additional line of processing to change 'companySummaryListRows' a bit more could just be encapsulated into the 'buildCompanySummaryListRows()' function? Would mean passing 'req' into too...

Comment (mwestacott) [inline src/controllers/check.your.answers.controller.ts:31]: Just wondering if you could encapsulate the logic that adds the e-mail JSON into the 'buildCompanySummaryListRows()' function? Pass in another optional 'email' parameter which is checked to see if JSON is added....

File: src/global/summary.list.rows.builder.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    {key: {html: "Registered office address"}, value: {html: companyAddress}},
+  );
+
+  if (company.accountsDue) {
+    let accountsOverdueHTML = company.accountsDue;
+    if (company.isAccountsOverdue || isDueDatePassed) {
+      accountsOverdueHTML = accountsOverdueHTML + "<br/><span class=\"govuk-!-font-weight-bold\">Your accounts are overdue</span>";
+    }
+    rowsArray = rowsArray.concat({key: {html: "Accounts due"}, value: {html: accountsOverdueHTML}});
+  }
```
Comment (mwestacott) [inline src/global/summary.list.rows.builder.ts:28]: See comment above - so potentially add the e-mail JSON section here if optional e-mail parameter is passed in and not null.

Comment (SVarallo) [reply src/global/summary.list.rows.builder.ts:31]: Done 👍

**Pull request merged**